app/components/directives/teacher-modal.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * Created by maddog on 3/4/2018.
+ */
+
+(
+    function(){
+        angular.module('MockApp')
+            .directive('teacherModal',teacherModal);
+
+        function teacherModal(){
+            var directive = {
+                restrict: 'EA',
+                scope:{
+                    'type': '@modalType',
+                    'title': '@buttonTitle',
+                    'class': '@buttonClass',
+                    'fileName': '@fileName',
+                    modalValue: '=',
+                },
+                template:'<a type="button" class="{{class}}" ng-click="modalDCtrl.open(modalValue,fileName)">{{title}}</a>',
+                controller:'modalDirectiveController as modalDCtrl',
+            }
+            return directive;
+        }
+
+        angular.module('MockApp')
+            .controller('modalDirectiveController', modalDirectiveController);
+
+        modalDirectiveController.$inject = ['$uibModal','TeacherService','$scope', '$rootScope'];
+
+        function modalDirectiveController($uibModal , TeacherService , $scope, $rootScope){
+            var vm = this;
+            $scope.message = null;
```

Comment:
what is the use of vm if you are going to set all variable from $scope. Please use vm instead of scope